File: week1-array-string/problem03-배열만들기2/heetae.py
# 문제: 배열 만들기 2
# 접근 방식: l부터 r까지 모든 수를 문자열로 바꾼 각 자릿수가 0 또는 5인지 확인한다.
# 시간복잡도: O((r - l + 1) log r) (각 수의 자릿수까지 검사)

def solution(l, r):
    answer = []

    for i in range(l, r + 1):
        valid = True

        for c in str(i):
            if c != '0' and c != '5':
                valid = False
                break

        if valid:
            answer.append(i)

    if not answer:
        return [-1]

    return answer

File: week3-hash-sort/problem02-폰켓몬/heetae.java
// 문제: 폰켓몬
// 접근 방식: 그냥 Set에다 다 넣어놓고 최대한 많은 종류를 넣어야하니 N/2마리보다 사이즈가 크면  return N/2, 작을경우 size()
// 시간복잡도: O(n)

import java.util.*;
class Solution {
    public int solution(int[] nums) {
        int answer = nums.length/2;
        
        Set<Integer> set = new HashSet<>();
        
        for(int i:nums){
            set.add(i);
        }
        
        return set.size()>answer? answer : set.size();
    }
}
